### PR TITLE
Prac_05 code review request

### PR DESCRIPTION
**Pull request by SarahWiese**
Hi @DaniRyland-Lawson
Please check formatting and functions for errors and logic. 
Thanks

### CODE REVIEW

**Comment by SarahWiese** [comment]
Hi @Mitchell Douglas 
Can you also have a look at my code for Prac_05.
Thanks

**Comment by SarahWiese** [comment]
@MdouglasJCU

see if this works

File: prac_05/word_occurences.py [hunk 1]
```diff
@@ -12,7 +12,17 @@ def main():
         else:
             sentence_dict[word] = 1
     for word, count in sentence_dict.items():
-        print("{:5} : {}".format(word, count))
+        print("{:{}} : {}".format(word, max_length_word(sentence), count))
+
+
+def max_length_word(sentence):
```
Comment (MdouglasJCU) [inline prac_05/word_occurences.py]: Able to simplify code, removing the need of an additional function.
max((len(word) for X in Y) 
Fill in X and Y.  You will also be able to remove a print statement in your for loop starting on line 14 once this has been done.

File: prac_05/word_occurences.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+"""Program to count occurrences of words in a string."""
+sentence = str(input("Enter sentence: ")).split()
```
Comment (MdouglasJCU) [inline prac_05/word_occurences.py]: No need to define str() for input(), as input by default is a string.

File: prac_05/hex_colour.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+"""Program to look up hexadecimal colour codes."""
+HEX_COLOURS = {"red1": "#ff0000", "purple": "#9020f0", "SpringGreen2": "#00ee76", "turquoise1": "#00f5ff", "yellow1": "#ffff00", "orchid1": "#ff83fa", "DeepSkyBlue1": "#00bfff", "chocolate1": "#ff7f24", "black": "#000000", "white": "#ffffff"}
+
+
+def main ():
+    """Program to return hexadecimal colour codes when colour name is entered."""
+    colour = str(input("Enter colour name: "))
+    while not is_colour_name_valid(colour):
```
Comment (MdouglasJCU) [inline prac_05/hex_colour.py]: You are able to heavily simplify this code by using a nested if statement within a while loop.  
Example:
colour = input
while colour != ""
if statement

**Review by MdouglasJCU — COMMENTED** [review]
Hi Sarah,
I've added a couple of comments for the hex colours and word occurrences.
State names looks great.

File: prac_05/hex_colour.py (hunk 3 above)
Comment (SarahWiese) [reply prac_05/hex_colour.py]: Thanks Mitchell. I've updated the code with your recommendations. I knew there had to be a better way just couldn't work it out.

File: prac_05/word_occurences.py (hunk 2 above)
Comment (SarahWiese) [reply prac_05/word_occurences.py]: Thanks. Updated wth change.

File: prac_05/word_occurences.py (hunk 1 above)
Comment (SarahWiese) [reply prac_05/word_occurences.py]: Updated program to correct print formatting. Thanks

File: prac_05/hex_colour.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    while colour != "":
+        if not is_colour_name_valid(colour):
+            print((HEX_COLOURS[colour]))
+            colour = str(input("Enter colour name:"))
+        else:
+            print("Invalid name!")
+            colour = str(input("Enter colour name:"))
+
+
+def is_colour_name_valid(colour):
```
Comment (lindsaymarkward) [inline prac_05/hex_colour.py:17]: This function is just one if and doesn't add a useful name that enhances the meaning.
I'd leave it out and just use:
`if colour in HEX_COLOURS.keys():` in main - it does the same thing and is readable on its own.

File: prac_05/state_names.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+while state != "":
+    if state in STATE_NAMES:
+        print(state, "is", STATE_NAMES[state])
+    else:
+        print("Invalid short state")
+    state = input("Enter short state: ").upper()
+
+# Print all state names.
+for short_state, long_state in STATE_NAMES.items():
+    print("{:3} is {:5}".format(short_state, long_state))
```
Comment (lindsaymarkward) [inline prac_05/state_names.py:21]: The `:5` here won't do anything as all long_states are longer than 5 characters.

File: prac_05/word_occurences.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+"""Program to count occurrences of words in a string and return list of words with their count."""
+
+
+def main():
+    """Program to count occurrences of words in a string."""
+    sentence = input("Enter sentence: ").split()
```
Comment (lindsaymarkward) [inline prac_05/word_occurences.py:6]: `sentence` is a good name for a sentence. A list of words would be called `words`.

File: prac_05/word_occurences.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    sentence.sort()
+    sentence_dict = {}
+    for word in sentence:
```
Comment (lindsaymarkward) [inline prac_05/word_occurences.py:9]: Remember "for singular in plural"?

File: prac_05/word_occurences.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+        if word in sentence_dict:
+            sentence_dict[word] += 1
+        else:
+            sentence_dict[word] = 1
+    for word, count in sentence_dict.items():
+        max_length = max((len(word) for word in sentence))
```
Comment (lindsaymarkward) [inline prac_05/word_occurences.py:15]: Why calculate this inside the loop? It's the same every time. So, if you have 100 words, you calculate the same maximum 99 times more than you need to.

**Pull request merged**